File: 2085-count-common-words-with-one-occurrence/2085-count-common-words-with-one-occurrence.cpp
class Solution {
public:
    int countWords(vector<string>& words1, vector<string>& words2) {
        // Use hashmap(unordered_map) to find count of words
        unordered_map<string, int> freq1, freq2;
        for (const auto& w : words1)
            ++freq1[w];
        for (const auto& w : words2)
            ++freq2[w];
        
        int ans = 0;
        // Find the common words and the both frequency of common words is 1.
        for (auto& [word, freq] : freq1){
            if (freq == 1 && freq2[word] == 1)
                ++ans;
        }
        
        return ans;
    }
};

File: 2085-count-common-words-with-one-occurrence/2085-count-common-words-with-one-occurrence.py
class Solution:
    def countWords(self, words1: List[str], words2: List[str]) -> int:
        # Use hash map to find count of words
        freq1, freq2 = Counter(words1), Counter(words2)
        ans = 0
        
        # Find the common words and its frequency is 1.
        for word, freq in freq1.items():
            if word in freq2 and freq == 1 and freq2[word] == 1:
                ans += 1
                
        return ans